### PR TITLE
Feat confetti onboarding #355

### PR DESCRIPTION
**Pull request by kamikazebr**
It create confetti effect on 
 - Onboardind screen.
 - Create proposal screen
Also some small fixes to workspace like envs and prettierrc.json.

https://user-images.githubusercontent.com/691510/144770464-db966268-2f3c-437f-98c2-ae57ba3d4f28.mp4

### CODE REVIEW

**Comment by kamikazebr** [comment]
Pls check if its correct, merging to #gardens-staging? And if have problem unrelated files like Dev and envs, prettier could be there or if need some new PR. @0xGabi

**Comment by kamikazebr** [comment]
# Conffeti on Proposal Screen
https://user-images.githubusercontent.com/691510/145872054-d42792d7-fb26-4290-a25e-07007be0025a.mp4

## Commit that
https://github.com/1Hive/gardens-ui/pull/729/commits/20eda2b558473e9e9e425214fd893ebc465e9074

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "singleQuote": true,
+  "semi": false,
+  "trailingComma": "es5",
+  "bracketSpacing": true,
+  "jsxBracketSameLine": false,
+  "tabSize": 2
+}
```
Comment (fabriziovigevani) [inline .prettierrc:8]: Note that we have the prettier rules under .eslintrc file with the prettier extension so we don´t nee this file

File: package.json [hunk 2]
```diff
@@ -4,15 +4,18 @@
   "private": true,
   "dependencies": {
     "3box": "^1.19.1",
+
     "@1hive/1hive-ui": "^1.0.14",
     "@1hive/connect-agreement": "^0.9.5",
     "@1hive/connect-gardens": "^0.1.12",
     "@1hive/connect-react": "^0.9.8",
     "@1hive/evmcrispr": "^0.3.2",
+    "@aragon/ui": "^1.9.3",
```
Comment (fabriziovigevani) [inline package.json:13]: Note that we are using `@1hive/1hive-ui` package not the  `@aragon/ui` one. Let´s remove this

File: src/components/Onboarding/Deployment/Boxes.js [hunk 3]
```diff
@@ -259,6 +279,7 @@ export function BoxReady({ isFinalized, onGetStarted, opacity, boxTransform }) {
         >
           {isFinalized ? (
             <div>
+              <PageVisibility onChange={throwConfetti} />
```
Comment (fabriziovigevani) [inline src/components/Onboarding/Deployment/Boxes.js:282]: Wondering why we use `PageVisibility` here and not for the create proposal screen. Also is it totally necessary?

File: src/providers/Gardens.js [hunk 4]
```diff
@@ -6,17 +6,15 @@ import React, {
...
 import useGardenFilters from '@hooks/useGardenFilters'
 import { useGardenRoute } from '@hooks/useRouting'
-import { useWallet } from './Wallet'
-
-import { fetchFileContent } from '../services/github'
-import { getVoidedGardensByNetwork } from '../voided-gardens'
 import { mergeGardenMetadata } from '@utils/garden-utils'
 import { testNameFilter } from '@utils/garden-filters-utils'
+import { fetchFileContent } from '../services/github'
+import { getVoidedGardensByNetwork } from '../voided-gardens'
```
Comment (fabriziovigevani) [inline src/providers/Gardens.js:17]: Why did you change the import orders here?

File: DEV GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+run 
+    yarn
+run 
+    yarn start
+
+Create an address on MetaMask.
+Use the Rinkeby TestNet
+
+Create an account on https://app.pinata.cloud/ and get the token on your env to REACT_APP_PINATA_API_TOKEN will convert to PINATA_API_TOKEN
+Select all property and get the JWT there
```
Comment (fabriziovigevani) [inline DEV GUIDE.md:15]: Let´s state that this is only necessary to test the onboarding process.

File: src/components/Garden/ModalFlows/CreateProposalScreens/CreateProposalScreens.js [hunk 6]
```diff
@@ -7,6 +7,7 @@ import { useAgreement } from '@hooks/useAgreement'
 import { useWallet } from '@providers/Wallet'
 import useActions from '@hooks/useActions'
 import { useStakingState } from '@providers/Staking'
+import confetti from 'canvas-confetti'
```
Comment (fabriziovigevani) [inline src/components/Garden/ModalFlows/CreateProposalScreens/CreateProposalScreens.js:10]: Let´s move this import to line 2.

We usually place external libraries imports first

File: src/components/Onboarding/Deployment/Boxes.js [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import React from 'react'
+import React, { useEffect, useCallback } from 'react'
```
Comment (fabriziovigevani) [inline src/components/Onboarding/Deployment/Boxes.js:1]:
```suggestion
import React, { useCallback, useEffect } from 'react'
```

**Review by fabriziovigevani — COMMENTED** [review]
Awesome! Nice work. 

Leaving some comment, let me know what do you think.

File: .prettierrc (hunk 1 above)
Comment (kamikazebr) [reply .prettierrc:8]: My VS dont read the rules there and break all the quotes in the code. I saw too @tiago on #ts-support will added that file.

Have some another way make my prettier of VSCode with the .eslintrc? i not waste some time on that, so i get the fast way to work.

File: package.json (hunk 2 above)
Comment (kamikazebr) [reply package.json:13]: Without it the `"sync-assets": "copy-aragon-ui-assets ./public",` copy-aragon-ui-assets failed. Need install it globally?

File: src/components/Onboarding/Deployment/Boxes.js (hunk 3 above)
Comment (kamikazebr) [reply src/components/Onboarding/Deployment/Boxes.js:282]: The create proposal screen have the autoclose modal, so dont wokr the same way like the Onboardind wait for the user interact to close the modal.

File: DEV GUIDE.md (hunk 5 above)
Comment (kamikazebr) [reply DEV GUIDE.md:15]: Perfect! Will change that.

File: src/providers/Gardens.js (hunk 4 above)
Comment (kamikazebr) [reply src/providers/Gardens.js:17]: Humm, I think it was why i REBASE to get the new commits and on the merge it get swap.
I'll fix that. Ty for get it.

File: src/components/Garden/ModalFlows/CreateProposalScreens/CreateProposalScreens.js (hunk 6 above)
Comment (kamikazebr) [reply src/components/Garden/ModalFlows/CreateProposalScreens/CreateProposalScreens.js:10]: Gotcha! That info need go to some CONTRIBUTING or DEV GUIDE?

File: DEV GUIDE.md (hunk 5 above)
Comment (kamikazebr) [reply DEV GUIDE.md:15]: So on that DEV Guide i told to install @aragon-ui too. So maybe its better get it DEV GUIDE off that PR, what you think?

**Comment by kamikazebr** [comment]
### Onboarding With Garden Theme Colors
https://user-images.githubusercontent.com/691510/145888640-8e178823-8c56-4613-be31-0bb268055243.mp4

File: src/components/Garden/ModalFlows/CreateProposalScreens/CreateProposalScreens.js (hunk 6 above)
Comment (0xGabi) [reply src/components/Garden/ModalFlows/CreateProposalScreens/CreateProposalScreens.js]: Yes, we use the following order for imports: `libraries > componentes > providers > hooks > utils > assets`.

Comment (kamikazebr) [reply src/components/Garden/ModalFlows/CreateProposalScreens/CreateProposalScreens.js]: Wow! Yah that definity need put on guide. I'll compile that and provide some examples.
I'll make it on some GIST to not mess up with review on that PR.

Or make new PR just to GUIDE

File: DEV GUIDE.md (hunk 5 above)
Comment (kamikazebr) [reply DEV GUIDE.md:15]: I'll make some PR to that to speed up that integration.

File: .prettierrc (hunk 1 above)
Comment (kamikazebr) [reply .prettierrc:8]: Will rename to same name of #738 so we get auto merge on that.

File: package.json (hunk 2 above)
Comment (kamikazebr) [reply package.json:13]: Removed from package.json. Very weird now it works.

File: src/components/Onboarding/Deployment/Boxes.js (hunk 3 above)
Comment (fabriziovigevani) [reply src/components/Onboarding/Deployment/Boxes.js:271]: What type of interaction is required in this case to throw the conffeti? Isn´t it thrown when the garden deployment si finalized as stated in the useEffect logic?

File: src/components/Garden/ModalFlows/CreateProposalScreens/CreateProposalScreens.js (hunk 6 above)
Comment (fabriziovigevani) [reply src/components/Garden/ModalFlows/CreateProposalScreens/CreateProposalScreens.js]: Yeah that makes more sense!

File: package.json (hunk 2 above)
Comment (fabriziovigevani) [reply package.json]: Actually the 1hive UI tollkit already provides this bin file since it´s a fork of arragon/ui

File: src/components/Onboarding/Deployment/Boxes.js (hunk 3 above)
Comment (kamikazebr) [reply src/components/Onboarding/Deployment/Boxes.js:271]: Why use page visibility ? 
Why signing is slow the whole process take time and confirm transactions, and you might been navigating on another tab of browser and just when your Wallet ask, you confirm signing. Or you are back and forth and if it finish and you not there you miss the confetti.

The use case for it was my self, while i was testing i miss the confetti because MetaMask ask to sign and I don’t need be there on that page (tab) to confirm.

So the use the PageVisibility its improve to UX, if yours think its not necessary i can remove it. @fabriziovigevani @0xGabi

File: .vscode/launch.json [hunk 8]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (reznord) [inline .vscode/launch.json:1]: Can you please remove this file and add `.vscode` to `.gitignore`? Because this is very specific to your local development.

**Review by reznord — DISMISSED** [review]

File: .vscode/launch.json (hunk 8 above)
Comment (kamikazebr) [reply .vscode/launch.json:1]: Ohh! Ofc! Tyvm for catch that.

Comment (kamikazebr) [reply .vscode/launch.json:1]: I think it already there. Its not mine commit. If we rebase from gardens to gardens-staging it will dissapear, you know why?

Comment (kamikazebr) [reply .vscode/launch.json:1]: I'll remove it anyway and add to gitignore

Comment (kamikazebr) [reply .vscode/launch.json:1]: It because some weird rebase i think.

Comment (kamikazebr) [reply .vscode/launch.json]: I reverted the merges made here and rebase, now looks great. @reznord

File: src/components/Onboarding/Deployment/Boxes.js (hunk 3 above)
Comment (0xGabi) [reply src/components/Onboarding/Deployment/Boxes.js:271]: Hey @kamikazebr what do you think of instead throwing the confetti on the `onClick` event of the `Get started` button a few lines below? 

It is the last step of the whole onboarding process seems the right time and we can remove the `react-page-visibility`.

**Review by 0xGabi — COMMENTED** [review]
I left a suggestion about the confetti for the onboaridng. Other than that LGTM! 🙌

File: src/components/Onboarding/Deployment/Boxes.js (hunk 3 above)
Comment (kamikazebr) [reply src/components/Onboarding/Deployment/Boxes.js:271]: Nice idea! I'll do that :D

File: .vscode/launch.json (hunk 8 above)
Comment (kamikazebr) [reply .vscode/launch.json]: Removed and added on gitignore

**Review by 0xGabi — APPROVED** [review]

**Comment by tmoutinho** [comment]
We should migrate this to ts 🔥
